packages/ui/src/DateTimeField.tsx

```diff
@@ -469,11 +504,19 @@ export const DateTimeField: React.FC<DateTimeFieldProps> = ({
                   shortTimezone
                   timezone={timezone}
                   onChange={(t) => {
-                    console.log("TimezonePicker onChange", t);
-                    setTimezone(t);
-                    // We need to call onBlur manually because the TimezonePicker doesn't support
-                    // it
-                    onBlur({timezone: t});
+                    if (onTimezoneChange) {
```

Comment:
I think this is reverting the timezone selection and it may very well be because we're not calculating isOnlyTimezoneChange correctly in the section above

Reply:
Huh, can you record a video? I'm not having issues (at least locally)

Reply:
https://www.loom.com/share/f13580086521499e9d65c78b3dc62e8a

This might actually be due to the timezone selector. I'm not seeing this issue in demo mode